digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitOperatorFactory.java

```diff
@@ -0,0 +1,167 @@
+package io.digdag.standards.operator.td;
+
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+import com.treasuredata.client.model.TDJob;
+import com.treasuredata.client.model.TDJobRequest;
+import com.treasuredata.client.model.TDJobRequestBuilder;
+import io.digdag.client.config.Config;
+import io.digdag.client.config.ConfigElement;
+import io.digdag.client.config.ConfigException;
+import io.digdag.spi.Operator;
+import io.digdag.spi.OperatorFactory;
+import io.digdag.spi.TaskExecutionException;
+import io.digdag.spi.TaskRequest;
+import io.digdag.spi.TaskResult;
+import io.digdag.spi.TemplateEngine;
+import io.digdag.util.BaseOperator;
+import org.msgpack.core.MessageTypeCastException;
+import org.msgpack.value.ArrayValue;
+import org.msgpack.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class TdWaitOperatorFactory
+        extends AbstractWaitOperatorFactory
+        implements OperatorFactory
+{
+    private static Logger logger = LoggerFactory.getLogger(TdWaitOperatorFactory.class);
+
+    private static final String JOB_ID = "jobId";
+
+    private final TemplateEngine templateEngine;
+
+    @Inject
+    public TdWaitOperatorFactory(TemplateEngine templateEngine, Config systemConfig)
+    {
+        super(systemConfig);
+        this.templateEngine = templateEngine;
+    }
+
+    public String getType()
+    {
+        return "td_wait";
+    }
+
+    @Override
+    public Operator newTaskExecutor(Path workspacePath, TaskRequest request)
+    {
+        return new TdWaitOperator(workspacePath, request);
+    }
+
+    private class TdWaitOperator
+            extends BaseOperator
+    {
+        private TdWaitOperator(Path workspacePath, TaskRequest request)
+        {
+            super(workspacePath, request);
+        }
+
+        @Override
+        public TaskResult runTask()
+        {
+            Config params = request.getConfig().mergeDefault(
+                    request.getConfig().getNestedOrGetEmpty("td"));
+
+            int pollInterval = getPollInterval(params);
+
+            Config state = request.getLastStateParams().deepCopy();
+
+            // Start the query job
+            Optional<String> existingJobId = state.getOptional(JOB_ID, String.class);
+            if (!existingJobId.isPresent()) {
+                String query = templateEngine.templateCommand(workspacePath, params, "query", UTF_8);
+                int rows = params.get("rows", int.class, 1);
+                String jobId = startJob(request, params, query, rows);
```

Comment:
how about moving startJob into following `try (TDOperator op = TDOperator.fromConfig(params)) {` block? Reason is that OperatorManager shows `Parameter '{}' is not used at task {}.` message if `params.get("engine", String.class, "presto")` is not called. TdWaitTableOperatorFactory also uses this way (existingJobId.isPresent is checked in `try (TDOperator op...` block).